llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp

```diff
@@ -1086,6 +1184,12 @@ X86MCCodeEmitter::emitVEXOpcodePrefix(int MemOperand, const MCInst &MI,
   case X86II::MRMSrcReg4VOp3: {
     // Instruction format for 4VOp3:
     //   src1(ModR/M), src2(ModR/M), src3(VEX_4V)
+    if ((TSFlags & X86II::EncodingMask) == X86II::EVEX) {
+      Prefix.setRR2(MI, CurOp++);
+      Prefix.setBB2(MI, CurOp++);
+      Prefix.set4V(MI, CurOp++);
+      break;
+    }
     Prefix.setR(MI, CurOp++);
```

Comment:
I found `bextr` also use this format, so need to set RR2?

Reply:
Oh, I got the answer. So we only set for APX case which must be EVEX encoding.

Reply:
Yes, you're right.